apps/server/src/shared/domain/domainobject/board/file-element.do.ts

```diff
@@ -0,0 +1,25 @@
+import { BoardComposite, BoardCompositeProps } from './board-composite.do';
+import type { AnyBoardDo } from './types';
+import type { BoardNodeBuildable } from './types/board-node-buildable';
+import type { BoardNodeBuilder } from './types/board-node-builder';
+
+export class FileElement extends BoardComposite implements FileElementProps, BoardNodeBuildable {
+	caption: string;
+
+	constructor(props: Omit<FileElementProps, 'children'>) {
+		super({ ...props, children: [] });
+		this.caption = props.caption;
+	}
+
+	isAllowedAsChild(): boolean {
```

Comment:
Without a parameter this method naming is even more confusing.
`fileElement.isAllowedAsChild();` 
I would understand that in the sense of an fileElement being allowed as child of a parent.
Therefore maybe something like this:
`fileElement.allowsAsChild();`

Reply:
Will resolve by Noth*

Reply:
Von Seitens der Base Klasse ließt sich das richtig, dort wird dann isAllowedAsChild bei add child als guard verwendet. 
ABER bei der Implementierung ist das für mich auch etwas verwirrend. --> FileElement.isAllowedAsChild(element)

Man könnte unterschiedliche Namings verwenden:
abstract XXX()
isAllowedAsChild = this.XXX

addChild() {
this.isAllowedAsChild()
}
@Metauriel